Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/type/TestCertificateContainer.kt

```diff
@@ -0,0 +1,114 @@
+package de.rki.coronawarnapp.coronatest.type
+
+import de.rki.coronawarnapp.covidcertificate.test.TestCertificate
+import de.rki.coronawarnapp.covidcertificate.test.TestCertificateData
+import de.rki.coronawarnapp.covidcertificate.test.TestCertificateQRCodeExtractor
+import de.rki.coronawarnapp.util.encryption.rsa.RSAKey
+import de.rki.coronawarnapp.vaccination.core.CertificatePersonIdentifier
+import de.rki.coronawarnapp.vaccination.core.personIdentifier
+import de.rki.coronawarnapp.vaccination.core.qrcode.QrCodeString
+import de.rki.coronawarnapp.vaccination.core.server.valueset.VaccinationValueSet
+import de.rki.coronawarnapp.vaccination.core.server.valueset.getDisplayText
+import okio.ByteString
+import org.joda.time.Instant
+import org.joda.time.LocalDate
+import java.util.Locale
+
+abstract class TestCertificateContainer {
+    abstract val identifier: TestCertificateIdentifier
+    abstract val registrationToken: RegistrationToken
+    abstract val type: CoronaTest.Type
+    abstract val registeredAt: Instant
+    abstract val publicKeyRegisteredAt: Instant?
+    abstract val rsaPublicKey: RSAKey.Public?
+    abstract val rsaPrivateKey: RSAKey.Private?
+    abstract val certificateReceivedAt: Instant?
+    abstract val encryptedDataEncryptionkey: ByteString?
+    abstract val encryptedDccCose: ByteString?
+    abstract val testCertificateQrCode: String?
+
+    abstract val isUpdatingData: Boolean
```

Comment:
Currently both test certificate types (RA/PCR) have the same attributes, so we re-use an abstract class, but still treat them individually, as this may change in the future and then we don't have to migrate any data structures.